### PR TITLE
Add com.github._0negal.Viper

### PR DESCRIPTION
**Pull request by GeckoEidechse**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
   **Non-standard permissions:**
   - ~~`--filesystem=host`: I'd love to restrict this further but the point of this application is to autodetect the Steam install location of the game _"Titanfall 2"_ and then access that location.~~
      ~~In order to do that it needs access to Steam configuration file that specifies the install locations which can be located at multiple places inside the user's home directory depending on the install method (deb vs Flatpak). Further it then needs access to the game's install location which can be anywhere in the filesystem.~~
   - `--filesystem=home`: Required to access the Steam library location config file (read-only) as well as the installed _"Titanfall 2"_ game itself (write access to download and add the mod).
   - `--filesystem=/media`: Alternative install location for games on Steam if they are installed on a SD card. Common on platforms like the SteamDeck.
   - `--filesystem=/mnt`: Alternative install location for games on Steam if they are installed a secondary drive (like an HDD due to storage limitations on the main drive.
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub.
     **Link:** https://github.com/0neGal/viper/issues/75#issuecomment-1090808769
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
    ~~There's a small issue with this part:~~
    ~~I cannot make the application ID 1:1 based on the repo URL as the owner's name starts with a zero, so the resulting ID would be `com.github.0neGal.viper`. However having an element in the ID start with a digit is disallowed by Flatpak. As such, [when discussing with the maintainer](https://github.com/0neGal/viper/issues/75#issuecomment-1077691898) we settled on `com.github.onegal.Viper`. Checking Github, there exists no user with the name `onegal` so I guess this should be fine ¯\\\_(ツ)\_/¯~~
    AppID updated to `com.github._0negal.Viper`
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.github.onegal.Viper.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+sdk: org.freedesktop.Sdk
+command: viper-run
+separate-locales: false
+tags: []
+finish-args:
+  - --socket=x11 # Allows us to create windows
+  - --share=ipc # Necessary for X11
+  - --share=network # Network access
+  - --device=dri # OpenGL rendering
+  - --filesystem=host # Full filesystem access, Titanfall2 could be installed anywhere on disk
```
Comment (hfiguiere) [inline com.github.onegal.Viper.yaml:15]: :-(

File: com.github.onegal.Viper.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        path: com.github.onegal.Viper.png
+      # Appdata XML for Flathub
+      - type: file
+        path: com.github.onegal.Viper.appdata.xml
+      # AppImage
+      - type: file
+        filename: Viper-1.3.0.AppImage
+        url: https://github.com/0neGal/viper/releases/download/v1.3.0/Viper-1.3.0.AppImage
+        sha256: 4e1342d99420b23ac4f6bfacf0ade34d2ec3768189b2d79cfc13fde403bc24fa
+        size: 103187559
```
Comment (hfiguiere) [inline com.github.onegal.Viper.yaml:57]:
```suggestion
```

File: com.github.onegal.Viper.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        path: com.github.onegal.Viper.desktop
+      # Icon
+      - type: file
+        path: com.github.onegal.Viper.png
+      # Appdata XML for Flathub
+      - type: file
+        path: com.github.onegal.Viper.appdata.xml
+      # AppImage
+      - type: file
+        filename: Viper-1.3.0.AppImage
```
Comment (hfiguiere) [inline com.github.onegal.Viper.yaml:54]:
```suggestion
```
this implicit

File: com.github.onegal.Viper.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
+app-id: com.github.onegal.Viper
+base: org.electronjs.Electron2.BaseApp
+base-version: '20.08'
+runtime: org.freedesktop.Platform
+runtime-version: '20.08'
+sdk: org.freedesktop.Sdk
+command: viper-run
+separate-locales: false
+tags: []
```
Comment (hfiguiere) [inline com.github.onegal.Viper.yaml:9]:
```suggestion
```

File: com.github.onegal.Viper.yaml [hunk 5]
```diff
@@ -0,0 +1,57 @@
+app-id: com.github.onegal.Viper
+base: org.electronjs.Electron2.BaseApp
+base-version: '20.08'
```
Comment (hfiguiere) [inline com.github.onegal.Viper.yaml:3]:
```suggestion
base-version: '21.08'
```

File: com.github.onegal.Viper.yaml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+runtime: org.freedesktop.Platform
+runtime-version: '20.08'
```
Comment (hfiguiere) [inline com.github.onegal.Viper.yaml:5]:
```suggestion
runtime-version: '21.08'
```

File: com.github.onegal.Viper.yaml [hunk 7]
```diff
@@ -0,0 +1,57 @@
+app-id: com.github.onegal.Viper
```
Comment (hfiguiere) [inline com.github.onegal.Viper.yaml:1]: It should be:

```suggestion
app-id: com.github._0negal.Viper
```


see https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-id-generic

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
bot, build com.github.onegal.viper

**Comment by hfiguiere** [comment]
bot, build com.github.onegal.Viper

File: com.github.onegal.Viper.yaml (hunk 7 above)
Comment (GeckoEidechse) [reply com.github.onegal.Viper.yaml:1]: Would manifest filename etc then be `com.github._0negal.Viper.yaml` or `com.github.0negal.Viper`? ^^

Comment (hfiguiere) [reply com.github.onegal.Viper.yaml:1]: yes

File: com.github.onegal.Viper.yaml (hunk 6 above)
Comment (GeckoEidechse) [reply com.github.onegal.Viper.yaml]: Updated in 6f3cce50d3e2b75e910d92a8c2b359f030ab55fb

File: com.github.onegal.Viper.yaml (hunk 5 above)
Comment (GeckoEidechse) [reply com.github.onegal.Viper.yaml]: Updated in 6f3cce50d3e2b75e910d92a8c2b359f030ab55fb

File: com.github.onegal.Viper.yaml (hunk 1 above)
Comment (GeckoEidechse) [reply com.github.onegal.Viper.yaml:15]: What's the issue here exactly? As explained in the initial description`--filesystem=host` is sadly necessary :/

File: com.github.onegal.Viper.yaml (hunk 2 above)
Comment (GeckoEidechse) [reply com.github.onegal.Viper.yaml:57]: `size` no longer needed?

Comment (hfiguiere) [reply com.github.onegal.Viper.yaml:55]: never has been

File: com.github.onegal.Viper.yaml (hunk 7 above)
Comment (GeckoEidechse) [reply com.github.onegal.Viper.yaml]: Done with 78de2a6e76af3d9ed929651bef253dfedf2935df

File: com.github.onegal.Viper.yaml (hunk 2 above)
Comment (GeckoEidechse) [reply com.github.onegal.Viper.yaml]: Removed with 5140b40e53b3a56acb8b36581926a8222c15fcae

**Comment by hfiguiere** [comment]
bot, build com.github._0negal.Viper

**Comment by GeckoEidechse** [comment]
Also I forgot to mention but if accepted, @0neGal should probably also be given write access to the repo ^^

File: README.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<p align="center">
```
Comment (barthalion) [inline README.md:1]: FWIW, users are unlikely to ever read any of this, so I'd say it can be removed altogether.

File: update-version.py [hunk 9]
```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
```
Comment (barthalion) [inline update-version.py:1]: How is this file going to be triggered, manually?

**Comment by barthalion** [comment]
> --filesystem=host: I'd love to restrict this further but the point of this application is to autodetect the Steam install location of the game "Titanfall 2" and then access that location.
In order to do that it needs access to Steam configuration file that specifies the install locations which can be located at multiple places inside the user's home directory depending on the install method (deb vs Flatpak). Further it then needs access to the game's install location which can be anywhere in the filesystem.

Please bear with me for a minute… If we're talking about installations done by host-level Steam or flatpaked Steam, wouldn't this location be the same in 99% of cases?

**Comment by GeckoEidechse** [comment]
> > --filesystem=host: I'd love to restrict this further but the point of this application is to autodetect the Steam install location of the game "Titanfall 2" and then access that location.
> > In order to do that it needs access to Steam configuration file that specifies the install locations which can be located at multiple places inside the user's home directory depending on the install method (deb vs Flatpak). Further it then needs access to the game's install location which can be anywhere in the filesystem.
> 
> Please bear with me for a minute… If we're talking about installations done by host-level Steam or flatpaked Steam, wouldn't this location be the same in 99% of cases? The point of my question is that we generally prefer locked-down permissions that are good enough for the majority of users over wide ones, as they can always be modified through CLI or Flatseal.

I agree, I'd also prefer more locked down permissions but the issue is the following:

So the location of the Steam library config file is known and is always within the user's home directory [in one of 3 places](https://github.com/0neGal/viper/blob/main/src/extras/findgame.js#L62-L64):
- `~/.steam/steam/steamapps/libraryfolders.vdf`
- `~/.var/app/com.valvesoftware.Steam/.steam/steam/steamapps/libraryfolders.vdf`
- `~/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/libraryfolders.vdf`


The **default** install location for the game itself is usually also within the user's home directory (depending on how Steam was installed):
- `~/.steam/steam/steamapps/common/`
- `~/.var/app/com.valvesoftware.Steam/.steam/steam/steamapps/common/`
- `~/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/`

However that's just the **default** location. For example one of our targets with Viper is the SteamDeck where the game it adds the mods for could be installed on the SD card. So we also need file system access to that location (usually `/media`?).

There's also users that have games installed on a secondary drive (usually some HDD) due to games' ever increasing filesize. As such to make the install as smooth as possible for those users, `/mnt` access would probably also be needed as well.

Then finally, to my knowledge, on Linux a user can add arbitrary locations in the filesystem as a _Steam library folder_ which is where things start to break down if we limit ourselves to just home directory, `/mnt` and `/media`.

Now I agree with your reasoning that 99% of users will either have their install in either of those 3 (`$HOME`, `/media`, `/mnt`), so I can try to update the manifest accordingly. However this means we'll also have to add some documentation somewhere for the last 1% of users to tell them how to enable full access via Flatseal.

File: update-version.py (hunk 9 above)
Comment (GeckoEidechse) [reply update-version.py:1]: Yes, it's triggered manually. This is a helper script to simply automatically update the manifest after upstream releases an update.

File: README.md (hunk 8 above)
Comment (GeckoEidechse) [reply README.md:1]: The whole README? I'd prefer to keep it for anyone who accidentally stumbles across the repo or address any Flatpak specific issues the application might have (like with custom game install locations as described below).

**Comment by GeckoEidechse** [comment]
I restricted file system access now in 8cc035d2ca675e1508f9ba2dfa3e6e433079d8c5

File: README.md (hunk 8 above)
Comment (GeckoEidechse) [reply README.md:1]: Oh I noticed I still had a download link in the header of readme that points to the installer of the Windows version.

I cleaned it up to only include the header image (as I initially intended) in 97d52283364de6e790a1461960bfcfdce669aee0

**Comment by GeckoEidechse** [comment]
bot, build com.github._0negal.Viper

**Comment by GeckoEidechse** [comment]
Newest build seems to still work ^^

**Pull request closed without merging**